sensor_height: 1.723
save_flag: true

patchwork:
    mode: "czm" 
    verbose: true # To check effect of uprightness/elevation/flatness
    visualize: true # Ground Likelihood Estimation is visualized
    # Ground Plane Fitting parameters
    num_iter: 3
    num_lpr: 20
    num_min_pts: 10
    th_seeds: 0.5
    th_dist: 0.125
    max_r: 80.0
    min_r: 2.7 # to consider vicinity of mobile plot form.
    uprightness_thr: 0.707 # 45: 0.707 / 60: 0.866

    adaptive_seed_selection_margin: -1.1 # The points below the adaptive_seed_selection_margin * sensor_height are filtered
    uniform: # deprecated
        num_rings: 16
        num_sectors: 54
    czm:
        # Note that `num_zones` == size of `num_sectors_each_zone` == size of `num_rings_each_zone` == size of `min_ranges` - 1
        # To divide zones, max_r, min_r, and min_ranges are utilized
        num_zones: 4
        num_sectors_each_zone: [16, 32 ,54, 32]
        num_rings_each_zone: [2, 4, 4, 4]
        # Note that `min_r` == `min_ranges_each_zone`[0]!
        min_ranges_each_zone: [2.7, 12.3625, 22.025, 41.35]
        # The elevation_thresholds are with respect to the sensor frame
        # Thus, actually (sensor_height - threshold_value) is the maximum ground height.
        # For instance, for the first ring, 1.723 - 1.2 = 0.523 is acceptable, maximum ground height
        # Note that the size of below parameters are indepentent to `num_zones`!
        elevation_thresholds:  [-1.2, -0.9984, -0.851, -0.605] # For elevation. The size should be equal to flatness_thresholds vector
        flatness_thresholds:  [0.0, 0.000125, 0.000185, 0.000185]  # For flatness. The size should be equal to elevation_thresholds vector
         


